Core/include/Acts/EventData/Track.hpp

```diff
@@ -8,794 +8,6 @@
-    return m_traj->trackStateRange(tip);
-  }
-
-  auto trackStateRange(IndexType itrack) const {
-    auto tip = component<IndexType, hashString("tipIndex")>(itrack);
-    return m_traj->trackStateRange(tip);
-  }
-
- private:
-  template <typename T, bool RO = ReadOnly, typename = std::enable_if_t<!RO>>
-  void copyDynamicFrom(IndexType dstIdx, const T& src, IndexType srcIdx) {
-    container().copyDynamicFrom_impl(dstIdx, src, srcIdx);
-  }
-
-  detail_tc::ConstIf<holder_t<track_container_t>, ReadOnly> m_container;
-  detail_tc::ConstIf<holder_t<traj_t>, ReadOnly> m_traj;
-};
-
-template <typename track_container_t, typename traj_t>
-TrackContainer(track_container_t& container, traj_t& traj)
-    -> TrackContainer<track_container_t, traj_t, detail::RefHolder>;
-
-template <typename track_container_t, typename traj_t>
-TrackContainer(track_container_t&& container, traj_t&& traj)
-    -> TrackContainer<track_container_t, traj_t, detail::ValueHolder>;
-
-/// Utility class that eases accessing dynamic columns in track containers
-/// @tparam T the type of the value to access
-/// @tparam ReadOnly true if this is a const accessor
-template <typename T, bool ReadOnly>
-struct TrackAccessorBase {
-  HashedString key;
-
-  /// Create the accessor from an already-hashed string key
-  /// @param _key the key
-  TrackAccessorBase(HashedString _key) : key{_key} {}
-  /// Create the accessor from a string key
-  /// @param _key the key
-  TrackAccessorBase(const std::string& _key) : key{hashString(_key)} {}
-
-  /// Access the stored key on the track given as an argument. Mutable version
-  /// @tparam track_proxy_t the type of the track proxy
-  /// @param track the track to access
-  /// @return mutable reference to the column behind the key
-  template <typename track_proxy_t, bool RO = ReadOnly,
-            typename = std::enable_if_t<!RO>>
-  T& operator()(track_proxy_t track) const {
-    static_assert(!track_proxy_t::ReadOnly,
-                  "Cannot get mutable ref for const track proxy");
-    return track.template component<T>(key);
-  }
-
-  /// Access the stored key on the track given as an argument. COnst version
-  /// @tparam track_proxy_t the type of the track proxy
-  /// @param track the track to access
-  /// @return const reference to the column behind the key
-  template <typename track_proxy_t, bool RO = ReadOnly,
-            typename = std::enable_if_t<RO>>
-  const T& operator()(track_proxy_t track) const {
-    if constexpr (track_proxy_t::ReadOnly) {
-      return track.template component<T>(key);
-    } else {
-      typename track_proxy_t::ConstTrackProxy ctrack{track};
-      return ctrack.template component<T>(key);
-    }
-  }
-};
-
-template <typename T>
-using TrackAccessor = TrackAccessorBase<T, false>;
-template <typename T>
-using ConstTrackAccessor = TrackAccessorBase<T, true>;
-
-}  // namespace Acts
+#include "Acts/EventData/TrackContainer.hpp"
```

Comment:
Keeping this header around for now, but printing a warning.

Reply:
is this to avoid breaking changes? do we have a followup for the next version on this?

Reply:
Yeah I wanted to keep the other header around for another version or so @andiwand.